install_microshift.py

```diff
@@ -0,0 +1,207 @@
+                break
+        else:
+            if 'FINISHED' in status_output:
+                break
+        logger.info("waiting on build to finish")
+        time.sleep(5)
+
+
+def microshift_iso_builder(h: host.Host, name_of_final_iso: str, version: str):
+    cleanup_microshift(h)
+    generate_kickstart()
+
+    packages = ["osbuild-composer", "composer-cli", "cockpit-composer", "bash-completion", "firewalld", "lorax"]
+    for package in packages:
+        install_packages(package, h)
+
+    cmd = "systemctl enable osbuild-composer.socket cockpit.socket --now"
+    output = h.run(cmd)
+
+    cmd = "uname -m"
+    uname_m = h.run(cmd).out.strip()
+
+    content_rhocp = f'''
+id = "rhocp-{version}"
+name = "Red Hat OpenShift Container Platform {version} for RHEL 9"
+type = "yum-baseurl
+url = "https://cdn.redhat.com/content/dist/layered/rhel9/{uname_m}/rhocp/{version}/os"
+check_gpg = true
+check_ssl = true
+system = false
+rhsm = true'''.strip()
+
+    rhocp_file = "rhocp-4.14.toml"
+    generate_toml_file(content_rhocp, rhocp_file, h)
+
+    content_fdp = f'''
+id = "fast-datapath"
+name = "Fast Datapath for RHEL 9"
+type = "yum-baseurl"
+url = "https://cdn.redhat.com/content/dist/layered/rhel9/{uname_m}/fast-datapath/os"
+check_gpg = true
+check_ssl = true
+system = false
+rhsm = true'''.strip()
+
+    fdp_file = "fast-datapath.toml"
+    generate_toml_file(content_fdp, fdp_file, h)
+
+    composer_cli_cmd("sources add", h, rhocp_file)
+    composer_cli_cmd("sources add", h, fdp_file)
+    composer_cli_cmd("blueprints push", h, "minimal-microshift.toml")
+
+    build_id = composer_cli_cmd(f"compose start-ostree --ref 'rhel/9/{uname_m}/edge' minimal-microshift edge-container", h).split(" ")[1]
+    logger.info(build_id)
+
+    wait_for_build_to_finish(h)
+
+    composer_cli_cmd(f"compose image {build_id}", h)
+    change_permissions(f"{build_id}-container.tar")
+    image_id_command = f"sudo podman load -i ./{build_id}-container.tar"
+
+    result = h.run(image_id_command)
+
+    image_id = result.out.split(':')[-1].strip()
+
+    cmd = f"sudo podman run -d --name=minimal-microshift-server -p 8085:8080 {image_id}"
+    result = h.run(cmd)
+
+    composer_cli_cmd("blueprints push", h, "microshift-installer.toml")
+
+    url = "http://127.0.0.1:8085/repo/"
+    ref = f"rhel/9/{uname_m}/edge"
+
+    build_command_two = composer_cli_cmd(f"compose start-ostree --url {url} --ref \"{ref}\" microshift-installer edge-installer", h)
+
+    build_id_two = build_command_two.split(" ")[1]
+
+    time.sleep(10)
+
+    wait_for_build_to_finish(h, build_id_two)
+
+    composer_cli_cmd(f"compose image {build_id_two}", h)
+
+    change_permissions(f"{build_id_two}-installer.iso")
+
+    vgrename(h)
+
+    generate_final_iso(f"{build_id_two}-installer.iso", name_of_final_iso, h)
+
+    shutil.rmtree("/var/cache/osbuild-worker/osbuild-store/stage/")
+
+    cmd = "systemctl disable osbuild-composer.socket cockpit.socket --now"
```

Comment:
enable doesn't do much until you reboot. Are you sure we need to enable and then disable?

Reply:
Could you elaborate on what you mean? I disabled due to this comment left above: https://github.com/bn222/cluster-deployment-automation/pull/95#discussion_r1490559416

Reply:
`systemctl enable ...` does it work without that line?

Reply:
I think it works without it. I can remove and double check